### PR TITLE
Fix handle comments and semicolons in SQL queries (#268)

### LINKED ISSUE #268: handle comment or semicolon at the query tail
### Describe the bug
see https://github.com/ClickHouse/clickhouse-rs/pull/267 - `;` or `--` at query tail is valid format

### PR DESCRIPTION
**Pull request by stdfox**
## Summary

This PR fixes issue #268 by properly handling comments and semicolons at the end of SQL queries.

Changes:

- Fixed FORMAT clause insertion to place it before trailing semicolons, comments and whitespace, not after
- Added tests for various edge cases with comments and semicolons

However, it does not support nested comments (like `"SELECT 1 /* outer /* nested */ comment */"`) at the moment.

Happy to discuss alternative approaches if there are concerns about this implementation.

## Checklist
Delete items not relevant to your PR:
- [x] Unit and integration tests covering the common scenarios were added
- [x] A human-readable description of the changes was provided so that we can include it in CHANGELOG later

Based on #267
Closes #268

### CODE REVIEW

File: src/sql/mod.rs [hunk 1]
```diff
@@ -135,14 +142,82 @@ impl SqlBuilder {
...
                 Ok(sql)
             }
             Self::Failed(err) => Err(Error::InvalidParams(err.into())),
         }
     }
 
+    fn remove_comments(sql: &str) -> String {
+        enum State {
+            Normal,
+            InString,
```
Comment (serprex) [inline src/sql/mod.rs]: Also need `InIdentifier`

**Comment by slvrtrn** [comment]
I wonder if this issue can be fixed by just:

1) stripping all trailing `;` chars and newlines until something else
2) appending `\nFORMAT ...` instead of `FORMAT ...`

i.e.

```sql
SELECT 1
 --comment FORMAT RowBinaryWithNamesAndTypes
```

becomes

```sql
SELECT 1
 --comment
FORMAT RowBinaryWithNamesAndTypes
```

and something like

```sql
SELECT 1;;;;;;;;;;;;;;
;;;
;;;;;; FORMAT RowBinaryWithNamesAndTypes
```

becomes

```sql
SELECT 1
FORMAT RowBinaryWithNamesAndTypes
```

**Comment by stdfox** [comment]
The current solution to the problem can be one of:
- Before inserting output_format, remove all or only trailing comments and symbols from the query;
- Finding the right offset to insert the output format.

The problem with removing comments is that in certain circumstances, the user may expect comments in the query ([as example](https://trino.io/docs/current/connector/clickhouse.html#appending-query-metadata)). If the implementation of `query_raw` or `interpolation flags` takes place, it will be possible to return the removal function as one of the builder options.


Thus, I implemented a simpler option (and I find it more preferable for now):
- The current sql builder code is minimally modified by adding only the offset search function for inserting output_format;
- Minimally changes the user's SQL code, preserving comments and line breaks;
- It covers most cases, including multi-line comments (not nested).

File: src/sql/mod.rs (hunk 1 above)
Comment (stdfox) [reply src/sql/mod.rs]: Thanks! Added InIdentifier. Nevertheless, the code is the same as InString. Is splitting it a good idea? Maybe merge it to InString(char)?

**Comment by slvrtrn** [comment]
> Before inserting output_format, remove all or only trailing comments and symbols from the query

I don't think we need to remove the trailing comments. We could just always insert the FORMAT clause with a newline symbol, as I mentioned in my previous comment. Other trailing symbols, e.g., semicolon, are safe to remove, though.

Please note that I am not opposed to the current solution cause this is great anyway, and I think it is worth keeping 👍 

So it's up to @serprex and @loyd

**Comment by slvrtrn** [comment]
@serprex, can you please have another look? Especially at the comments starting from https://github.com/ClickHouse/clickhouse-rs/pull/269#issuecomment-3154890853

**Comment by slvrtrn** [comment]
Closed in favor of #311
